src/analyzing/CFG_build.ml

```diff
@@ -240,41 +259,122 @@ let rec cfg_stmt : state -> F.nodei option -> stmt -> cfg_stmt_result =
   | Return (tok, e) ->
       let newi = state.g#add_node { F.n = F.NReturn (tok, e) } in
       state.g |> add_arc_from_opt (previ, newi);
-      state.g |> add_arc (newi, state.exiti);
+      state.g |> add_arc (newi, state.return_destination);
+      state.may_return := true;
```

Comment:
What happens if you have this code:

```python
if something:
  return;
try:
  ...
```
The `return;` will set `may_return` to true even though it's not inside the `try`, right? How would that affect the CFG of the `try`?

Reply:
Thanks for catching this bug! It will be fixed in the next update.